crates/bevy_diagnostic/src/system_diagnostics.rs

```diff
@@ -0,0 +1,116 @@
+use std::any::TypeId;
+
+use super::{Diagnostic, DiagnosticId, Diagnostics};
+use bevy_ecs::{
+    ArchetypeComponent, Resources, System, SystemId, ThreadLocalExecution, TypeAccess, World,
+};
+use bevy_utils::Instant;
+
+pub trait MeasuredSystemExt<In, Out>: System<In = In, Out = Out> + Sized {
+    /// Add a system and record its execution time in [Diagnostics].
+    ///
+    /// # Example
+    /// ```
+    /// use bevy::{
```

Comment:
this will fail when running doctest because you can't import `bevy` from inside one of the subcrates, you have to import the one you actually use without the global prelude

Reply:
I intended to show how to use it from outside of bevy, but I hadn't thought about doc tests. One day I will remember everything needed to pass the checks (maybe)